pkg/tls/certRenewer.go

```diff
@@ -117,8 +118,22 @@ func (c *CertRenewer) WriteCACertToSecret(caPEM *PemPair, props CertificateProps
 	logger := c.log.WithName("CAcert")
 	name := generateRootCASecretName(props)
 
-	secretUnstr, err := c.client.GetResource("", "Secret", props.Namespace, name)
+	depl, err := c.client.GetResource("", "Deployment", props.Namespace, "kyverno")
+
+	rsHash := ""
 	if err != nil {
+		rsHash = fmt.Sprintf("%v", depl.GetUID())
+	}
+
+	var rsHashSec string = "default"
+	var ok bool
+
+	secretUnstr, err := c.client.GetResource("", "Secret", props.Namespace, name)
+	if err == nil {
+		rsHashSec, ok = secretUnstr.GetAnnotations()[MasterDeploymentUID]
```

Comment:
Will this annotation be added to the custom certificate's secret?

Reply:
To, `kyverno-svc.kyverno.svc.kyverno-tls-ca`, yes.